File: VideoStreamingBackend/src/domain/repositories.ts
/**
 * Repository interfaces (ports) for the domain layer.
 */
import { StreamSession } from './entities/StreamSession';
import { ViewerSession } from './entities/ViewerSession';

export interface IStreamSessionRepository {
  get(cameraId: string): StreamSession | undefined;
  getAll(): StreamSession[];
  save(session: StreamSession): void;
  remove(cameraId: string): void;
}

export interface IViewerSessionRepository {
  get(viewerId: string): ViewerSession | undefined;
  getByCamera(cameraId: string): ViewerSession[];
  getByUser(userId: string): ViewerSession[];
  save(session: ViewerSession): void;
  remove(viewerId: string): void;
  removeByCamera(cameraId: string): void;
}

File: StreamingBackend/internal/domain/repositories/repositories.go
package repositories

import "github.com/vigileye/streaming-backend/internal/domain/entities"

// StreamSessionRepository defines the persistence contract for stream sessions.
type StreamSessionRepository interface {
	Get(cameraID string) (*entities.StreamSession, bool)
	GetAll() []*entities.StreamSession
	Save(session *entities.StreamSession)
	Remove(cameraID string)
}

// ViewerSessionRepository defines the persistence contract for viewer sessions.
type ViewerSessionRepository interface {
	Get(viewerID string) (*entities.ViewerSession, bool)
	GetByCamera(cameraID string) []*entities.ViewerSession
	GetByUser(userID string) []*entities.ViewerSession
	Save(session *entities.ViewerSession)
	Remove(viewerID string)
	RemoveByCamera(cameraID string)
}
